File: examples/rust/transport_queue_policy.rs
//! Tune queue policy for a Wyrd transport.
//!
//! Run with:
//!     cargo run -p wyrd-rust-examples --bin transport_queue_policy

use wyrd_client::transport::{MockConfig, QueueConfig, TransportConfig};

fn main() -> anyhow::Result<()> {
    let queue = QueueConfig {
        transport: TransportConfig::Mock(MockConfig {
            label: "queue-policy-demo".to_string(),
            fail_on_flush: None,
        }),
        flush_max_rows: 5_000,
        flush_interval_ms: 1_000,
        channel_capacity: 500,
        sample_ratio: Some(0.5),
    };
    queue.validate()?;

    let json = serde_json::to_string_pretty(&queue)?;
    let round_trip: QueueConfig = serde_json::from_str(&json)?;
    anyhow::ensure!(round_trip == queue, "queue config did not round-trip");
    println!("{json}");
    Ok(())
}

File: python/py-wyrd/examples/transport_queue_policy.py
"""Tune queue policy as JSON-compatible config.

Run with:
    python python/py-wyrd/examples/transport_queue_policy.py
"""

from __future__ import annotations

import json


def main() -> None:
    queue = {
        "transport": {
            "transport": "mock",
            "params": {
                "label": "queue-policy-demo",
                "fail_on_flush": None,
            },
        },
        "flush_max_rows": 5_000,
        "flush_interval_ms": 1_000,
        "channel_capacity": 500,
        "sample_ratio": 0.5,
    }
    encoded = json.dumps(queue, indent=2, sort_keys=True)
    assert json.loads(encoded) == queue
    print(encoded)


if __name__ == "__main__":
    main()
